### PR TITLE
BCDA-1597 Feature: 1800 Medicare ETL Process

### PR DESCRIPTION
**Pull request by tbellj**
Fixes [BCDA-1597](https://jira.cms.gov/browse/BCDA-1597)

Problem:
This is needed to successfully ingest 1-800 medicare suppression files.

Proposed changes:
ETL process for ingesting 1-800 medicare files, and the necessary db models.

Change Details:
bcda/models/models.go - modified the suppression struct, added the suppression file struct.
bcda/suppression/suppression.go - ETL process for ingesting 1-800 medicare files.
db/api.sql - added the suppression_files table, modified suppressions table to be more in line with our cclf data model.

Security Implications:
This change **will** deal with PII/PHI directly.
This change **does not** add new software dependencies
This change **does not** alter security controls or supporting software
This change **does** introduce storage of new data
- We will now be ingesting the contents of the 1-800 medicare files.
A security checklist **has** been completed for this change

Acceptance Validation:
Tests pass, but this is not considered fully complete until the synthetic file is verified with the ICD. Once that is done, then we can run a test ingest with the verified synthetic file and validate the contents in the db.

Screenshot:
screenshot_suppression

Feedback Requested:
Any and all

### CODE REVIEW

File: bcda/models/models.go [hunk 1]
```diff
@@ -374,17 +374,25 @@ type CCLFBeneficiary struct {
 
 type Suppression struct {
 	gorm.Model
+	CCLFFile            CCLFFile
```
Comment (embh) [inline bcda/models/models.go]:
```suggestion
	SuppressionFile            SuppressionFile
```

Comment (tbellj) [reply bcda/models/models.go]: eyes!!!! good catch

File: bcda/suppression/suppression_test.go [hunk 2]
```diff
@@ -0,0 +1,323 @@
...
+	metadata.filePath = metadata.filePath + "/blah/"
+	err = validate(metadata)
+	assert.NotNil(err)
+	assert.Contains(err.Error(), "could not read file "+metadata.filePath)
+
+	// invalid file header
+	metadata.filePath = BASE_FILE_PATH + "suppressionfile_BadHeader/T#EFT.ON.ACO.NGD1800.DPRF.D181120.T1000009"
+	err = validate(metadata)
+	assert.NotNil(err)
+	assert.EqualError(err, "invalid file header for file: "+metadata.filePath)
```
Comment (embh) [inline bcda/suppression/suppression_test.go]: `assert.EqualError()` checks for the error being non-nil, so we can leave out `assert.NotNil()` before it. https://godoc.org/github.com/stretchr/testify/assert#EqualError

File: bcda/suppression/suppression_test.go [hunk 3]
```diff
@@ -0,0 +1,323 @@
...
+
+func (s *SuppressionTestSuite) TestParseMetadata_InvalidFilename() {
+	assert := assert.New(s.T())
+
+	// invalid file name
+	_, err := parseMetadata("/path/to/file")
+	assert.EqualError(err, "invalid filename for file: /path/to/file")
+
+	_, err = parseMetadata("/path/T#EFT.ON.ACO.NGD1800.FRPD.D191220.T1000010")
+	assert.NotNil(err)
```
Comment (embh) [inline bcda/suppression/suppression_test.go]: Would be great to check the error message here.

File: bcda/suppression/suppression_test.go [hunk 4]
```diff
@@ -0,0 +1,323 @@
...
+	suppresslist = []suppressionFileMetadata{}
+	skipped = 0
+	err = filepath.Walk(folderPath, getSuppressionFileMetadata(&suppresslist, &skipped))
+	assert.Nil(err)
+	assert.Equal(0, len(suppresslist))
+	assert.Equal(2, skipped)
+
+	// assert that this file is not still here.
+	_, err = os.Open(filePath)
+	assert.NotNil(err)
```
Comment (embh) [inline bcda/suppression/suppression_test.go]: Could we check the error message here too?

**Pull request merged**